geode-core/src/main/java/org/apache/geode/internal/cache/tier/sockets/ServerConnection.java

```diff
@@ -1308,32 +1308,21 @@ protected int getClientReadTimeout() {
   }
 
   void setProcessingMessage() {
-    synchronized (processingMessageLock) {
-      // go ahead and reset it if it is already set
-      processingMessageStartTime = System.currentTimeMillis();
-    }
+    // go ahead and reset it if it is already set
+    processingMessageStartTime.set(System.currentTimeMillis());
   }
 
   void updateProcessingMessage() {
-    synchronized (processingMessageLock) {
-      // only update it if it was already set by setProcessingMessage
-      if (processingMessageStartTime != -1) {
-        processingMessageStartTime = System.currentTimeMillis();
-      }
-    }
+    // only update it if it was already set by setProcessingMessage
+    processingMessageStartTime.compareAndSet(-1, System.currentTimeMillis());
```

Comment:
This looks like you flipped the old "-1" logic. I think setting it to -1 in the old code, basically disabled updateProcessingMessage. If we are going to stick with that logic then:
this code should read it and if it is -1 return. Otherwise it can call compareAndSet only setting it if it is the oldValue. If it is not the old value I don't think it needs to spin since either some other thread concurrently set it to -1 (in which case this method is disabled) or to the current time which is what we were trying to do anyway. So no need for a loop. We just need to be careful to allow "-1" to have priority over

Reply:
Boo, thanks for catching that. I’m going to see if I can get this unit testable and try again.